### DISCUSSION TITLE
utility function of an agent

### DISCUSSION THREAD

**Discussion by YueNing** [yasserfarouk/scml#25] [one-shot-qa] [answered]
Definition of utility function of an agent in [full description](http://www.yasserm.com/scml/scml2021oneshot.pdf)

<img src="https://render.githubusercontent.com/render/math?math=u_{a}\left(C^{i}, C^{o}\right)=\sum_{c \in \bar{C}^{o}}\left(p_{c} q_{c}\right)-\sum_{c \in C^{i}}\left(p_{c} q_{c}\right)-m_{a} P_{a}-\gamma_{a} tp\left(p_{a}^{i}, d\right) Q_{a}^{i+}-\alpha_{a} tp\left(p_{a}^{o}, d\right) Q_{a}^{o+}">

Have a question about the definition of <img src="https://render.githubusercontent.com/render/math?math=\alpha_{a}"> and <img src="https://render.githubusercontent.com/render/math?math=\gamma_{a}">

First confirm the meaning of following two terms: 
- **Disposal cost or garbage collection:**  <img src="https://render.githubusercontent.com/render/math?math=\gamma_{a} \mathbf{t} \mathbf{p}\left(p_{a}^{i}, d\right) Q_{a}^{i+}"> ![gamma.PNG](https://i.loli.net/2021/05/01/aKNIvWoHZFhXjiy.png) a little confused about the sentence with red line. Doesn't whole term mean the **Disposal cost**?  I am not sure, whether i understand correctly.
- **Penalty the factory has to pay for failure to deliver a unit of its output product:** <img src="https://render.githubusercontent.com/render/math?math=\alpha_{a} \mathbf{t} \mathbf{p}\left(p_{a}^{o}, d\right) Q_{a}^{o+}"> 

Hence, <img src="https://render.githubusercontent.com/render/math?math=\alpha_{a}"> and <img src="https://render.githubusercontent.com/render/math?math=\gamma_{a}"> are similar to the meaning  of discount factor?

**Comment by yasserfarouk** [comment, marked as answer]
> Does not the whole term represent the disposal cost.

You are right. The whole term represents the disposal cost (we sometimes also call it garbage collection cost). $\gamma$ is the *multiplication factor* by which the trading price is multiplied to give the disposal cost per item.

> Penalty term 

The shortfall penalty is what the factory needs to pay for failure to deliver a unit of its output product. Correct. Nevertheless, I am not sure how does this relates to discounting.

**Reply by YueNing** [reply]
thanks for the explanation.
